### PR TITLE
apps/smartfs_powercut : Add auto powercut test app

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
Add application for auto testing.
It use all apis which handle  entry of smartfs.
Also to check integrity of journal, use 2 size of buffer, one is small(st_blksize) one is large

### CODE REVIEW

**Comment by amogh-samsung** [comment]
I think most operations have been included.
But currently all write operations are in multiples of the sector size.
How about writing bytes less than size of one sector.
And then appending to the file further?

File: apps/examples/smartfs_powercut/smartfs_powercut_main.c [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+#include <sys/ioctl.h>
+#include <sys/stat.h>
+
+#include <tinyara/fs/ioctl.h>
+#include <tinyara/config.h>
+
+/* Below paths are for test file which use small size of buffer(same as st_blksize) of stat */
+#define TEST_SMALL				"/mnt/test_small"
+#define TEST_SMALL_BACKUP		"/mnt/backup_small/test_small"
+#define TEST_SMALL_BACKUP_DIR	"/mnt/backup_small"
```
[the remark this answers is not in the document]
Comment (Taejun-Kwon) [reply apps/examples/smartfs_powercut/smartfs_powercut_main.c:74]: ok

File: apps/examples/smartfs_powercut/smartfs_powercut_main.c [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+
+#define TEST_LARGE				"/mnt/test_large"
+#define TEST_LARGE_BACKUP		"/mnt/backup_large/test_large"
+#define TEST_LARGE_BACKUP_DIR	"/mnt/backup_large"
```
[the remark this answers is not in the document]
Comment (Taejun-Kwon) [reply apps/examples/smartfs_powercut/smartfs_powercut_main.c:78]: ok

File: apps/examples/smartfs_powercut/smartfs_powercut_main.c [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+
+
+#define TEST_LARGE_BUFSIZE		8192
+int TEST_SMALL_BUFSIZE;
+
+/* Write Time for each file */
+#define TEST_WRITECOUNT			30
```
[the remark this answers is not in the document]
Comment (Taejun-Kwon) [reply apps/examples/smartfs_powercut/smartfs_powercut_main.c:85]: ok

**Comment by Taejun-Kwon** [comment]
> I think most operations have been included.
> But currently all write operations are in multiples of the sector size.
> How about writing bytes less than size of one sector.
> And then appending to the file further?

Good idea. Please add verysmall_file later  =D

File: apps/examples/smartfs_powercut/smartfs_powercut_main.c [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+
+	ret = create_file(TEST_LARGE_BACKUP, TEST_LARGE_BUFSIZE);
+	if (ret != OK && errno != EEXIST) {
+		return ERROR;
+	}
+	printf("create test file success!!\n");
+	return OK;
+}
+
+int do_test(char *src, char *backup, char *backupdir, int bufsize)
```
Comment (jeongchanKim) [inline apps/examples/smartfs_powercut/smartfs_powercut_main.c:339]: I think you'd better make this function as static.

File: apps/examples/smartfs_powercut/smartfs_powercut_main.c [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+
+/* Write Time for each file */
+#define TEST_WRITECOUNT         30
+
+struct buffer_data_s {
+	uint16_t crc;
+	char data[TEST_LARGE_BUFSIZE];
+};
+
+struct buffer_data_s g_buf;
```
Comment (jeongchanKim) [inline apps/examples/smartfs_powercut/smartfs_powercut_main.c:92]: Please declare this as static.

File: apps/examples/smartfs_powercut/smartfs_powercut_main.c [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+
+void make_random_chars(char *buf, size_t len)
+{
+	for (int i = 0; i < len - 1; i++) {
+		buf[i] = rand() % len;
+	}
+	buf[len - 1] = '\0';
+}
+
+int write_file(int fd, size_t bufsize)
```
Comment (jeongchanKim) [inline apps/examples/smartfs_powercut/smartfs_powercut_main.c:102]: This function name can be easily used, so please make static.

**Pull request merged**